test/rebar_dir_SUITE.erl

```diff
@@ -162,3 +174,23 @@ alt_base_dir_rel(Config) ->
     ?assert(filelib:is_dir(filename:join([BaseDir, "lib", Name2, "ebin"]))),
     ?assert(filelib:is_file(filename:join([BaseDir, "lib", Name2, "ebin", Name2++".app"]))),
     ?assert(filelib:is_file(filename:join([BaseDir, "lib", Name2, "ebin", Name2++".beam"]))).
+
+global_cache_dir(Config) ->
+    RebarConfig = [{erl_opts, []}],
+    {ok, State} = rebar_test_utils:run_and_check(Config, RebarConfig, ["compile"], return),
+    DataDir = ?config(priv_dir, Config),
+    Expected = filename:join([DataDir, "cache"]),
+    ?assertEqual(Expected, rebar_dir:global_cache_dir(rebar_state:opts(State))).
+
+default_global_cache_dir(Config) ->
+    RebarConfig = [{erl_opts, []}],
+    {ok, State} = rebar_test_utils:run_and_check(Config, RebarConfig, ["compile"], return),
+    Expected = filename:join([rebar_dir:home_dir(), ".cache", "rebar3"]),
+    ?assertEqual(Expected, rebar_dir:global_cache_dir(rebar_state:opts(State))).
+
+overwrite_default_global_cache_dir(Config) ->
+    os:putenv("REBAR_CACHE_DIR", "/tmp/"),
```

Comment:
I'd rather we use `?config(priv_dir, Config)` for these paths so that they are written in the scratch directory for the test suite, avoiding any clash with other paths user may legitimately choose.